libs/application-generic/src/logging/index.ts

```diff
@@ -10,135 +11,85 @@ export function getErrorInterceptor(): NestInterceptor {
+    logLevel = process.env.LOGGING_LEVEL || process.env.LOG_LEVEL;
+  } else {
+    console.log(`Environment variable LOG_LEVEL is not set. Falling back to info level.`);
     logLevel = 'info';
   }
-  // eslint-disable-next-line no-console
-  console.log(`Log Level Chosen: ${logLevel}`);
-
-  return logLevel;
-}
-
-// TODO: should be moved into a config framework
-function getLoggingVariables(): ILoggingVariables {
-  const env = process.env.NODE_ENV ?? 'local';
-
-  // eslint-disable-next-line no-console
-  console.log(`Environment: ${env}`);
-
-  const hostingPlatform = process.env.HOSTING_PLATFORM ?? 'Docker';
-
-  // eslint-disable-next-line no-console
-  console.log(`Platform: ${hostingPlatform}`);
 
-  const tenant = process.env.TENANT ?? 'OS';
+  if (!loggingLevelArr.includes(logLevel)) {
+    console.log(`${logLevel}is not a valid log level of ${loggingLevelArr}. Falling back to info level.`);
 
-  // eslint-disable-next-line no-console
-  console.log(`Tenant: ${tenant}`);
+    return 'info';
+  }
 
-  return {
-    env,
-    level: getLogLevel(),
-    hostingPlatform,
-    tenant,
-  };
+  return logLevel;
 }
 
 export function createNestLoggingModuleOptions(settings: ILoggerSettings): Params {
-  const values: ILoggingVariables = getLoggingVariables();
-
-  let redactFields: string[] = sensitiveFields.map((val) => val);
-
+  let redactFields: string[] = sensitiveFields;
   redactFields.push('req.headers.authorization');
-
   const baseWildCards = '*.';
   const baseArrayWildCards = '*[*].';
   for (let i = 1; i <= 6; i += 1) {
     redactFields = redactFields.concat(sensitiveFields.map((val) => baseWildCards.repeat(i) + val));
-
     redactFields = redactFields.concat(sensitiveFields.map((val) => baseArrayWildCards.repeat(i) + val));
   }
 
-  const transport = ['local', 'test', 'debug'].includes(process.env.NODE_ENV) ? { target: 'pino-pretty' } : undefined;
-
-  // eslint-disable-next-line no-console
-  console.log(loggingLevelSet);
-
-  // eslint-disable-next-line no-console
-  console.log(`Selected Log Transport ${!transport ? 'None' : 'pino-pretty'}`, loggingLevelSet);
+  const configSet = {
+    transport: ['local', 'test', 'debug'].includes(process.env.NODE_ENV) ? { target: 'pino-pretty' } : undefined,
+    platform: process.env.HOSTING_PLATFORM ?? 'Docker',
+    tenant: process.env.TENANT ?? 'OS',
+    level: getLogLevel(),
+    levels: loggingLevelSet,
+  };
+  console.log('Logging Configuration:', {
+    level: configSet.level,
+    environment: process.env.NODE_ENV,
+    transport: !configSet.transport ? 'None' : 'pino-pretty',
+    platform: configSet.platform,
+    tenant: configSet.tenant,
+    levels: JSON.stringify(configSet.levels),
+  });
 
   return {
     exclude: [{ path: '*/health-check', method: RequestMethod.GET }],
+    assignResponse: true,
     pinoHttp: {
-      customLevels: loggingLevelSet,
-      level: values.level,
+      useOnlyCustomLevels: true,
```

Comment:
in the end i kept the custom levels, this flag is making sure that they are not overridden by the built-in levels in pino